.github/workflows/benchmarks.yml

```
@@ -0,0 +1,149 @@
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# Workflow for all benchmark-related jobs. It is designed to be called from
+# the main workflow ci.yml. The concurrency of this workflow is controlled by
+# the caller's job.
+
+name: Benchmarks
+
+on:
+  workflow_call:
+    inputs:
+      runner-group:
+        required: true
+        type: string
+      runner-env:
+        required: true
+        type: string
+      gcs-dir:
+        required: true
+        type: string
+      build-dir:
+        required: true
+        type: string
+      build-dir-archive:
+        required: true
+        type: string
+      build-dir-gcs-artifact:
+        required: true
+        type: string
+      tf-binaries-dir:
+        required: true
+        type: string
+      tf-binaries-archive:
+        required: true
+        type: string
+      tf-binaries-gcs-artifact:
+        required: true
+        type: string
+
+jobs:
+  build_benchmark_suites:
+    runs-on:
+      # Hacks, and order matters. See the comment at the top of the ci.yml.
+      - self-hosted
+      - runner-group=${{ inputs.runner-group }}
+      - environment=${{ inputs.runner-env }}
+      - cpu
+      - os-family=Linux
+    env:
+      BUILD_DIR: ${{ inputs.build-dir }}
+      BUILD_DIR_ARCHIVE: ${{ inputs.build-dir-archive }}
+      BUILD_DIR_GCS_ARTIFACT: ${{ inputs.build-dir-gcs-artifact }}
+      TF_BINARIES_DIR: ${{ inputs.tf-binaries-dir }}
+      TF_BINARIES_ARCHIVE: ${{ inputs.tf-binaries-archive }}
+      TF_BINARIES_GCS_ARTIFACT: ${{ inputs.tf-binaries-gcs-artifact }}
+    steps:
+      - name: "Checking out repository"
+        uses: actions/checkout@7884fcad6b5d53d10323aee724dc68d8b9096a2e # v2
+        with:
+          submodules: true
+      - name: "Downloading build dir archive"
+        run: gcloud alpha storage cp "${BUILD_DIR_GCS_ARTIFACT}" "${BUILD_DIR_ARCHIVE}"
+      - name: "Extracting install from build dir archive"
+        run: tar -xf "${BUILD_DIR_ARCHIVE}" "${BUILD_DIR}/install"
+      - name: "Downloading TF binaries archive"
+        run: gcloud alpha storage cp "${TF_BINARIES_GCS_ARTIFACT}" "${TF_BINARIES_ARCHIVE}"
+      - name: "Extracting TF binaries archive"
+        run: tar -xf "${TF_BINARIES_ARCHIVE}"
+      - name: "Building benchmarks"
+        run: |
+          build_tools/github_actions/docker_run.sh \
+            --env "IREE_TF_BINARIES_DIR=${TF_BINARIES_DIR}" \
+            --env "IREE_HOST_BINARY_ROOT=${BUILD_DIR}/install" \
+            gcr.io/iree-oss/base@sha256:5d43683c6b50aebe1fca6c85f2012f3b0fa153bf4dd268e8767b619b1891423a \
+            build_tools/cmake/build_benchmarks.sh
+
+  build_benchmark_tools:
+    runs-on:
+      # Hacks, and order matters. See the comment at the top of the ci.yml.
+      - self-hosted
```

Ditto comments above

```suggestion
  build_tools:
    runs-on:
      - self-hosted  # must come first
```